docs/templates/template_reference.md

```diff
@@ -521,7 +521,7 @@ The only way to remediate is to recompile and reinstall the kernel, so no remedi
 
     -   **type** - filesystem type. Used only in Bash remediation.
 
-    -   **mount_has_to_exist** - Specifies if the **mountpoint**
+    -   **fstab_entry_is_optional** - Specifies if the **mountpoint**
```

Comment:
I think the issue here is that the variable is not used only in the remediation as the description suggests, but also in OVAL, where the name doesn't match the logic.

In OVAL, if `mount_has_to_exist == True`, the check will pass when the mount does *not* exist (opposite to what one might expect from the name of the variable). That's really the reason behind this proposal.

In the remediation, I agree that the existing name makes sense intuitively, but only if you know the full context of the remediation. If `mount_has_to_exist == True`, it's not too wild to expect that the remediation would force-create a mount, since it seemingly has to exist. But the remediation only really uses this variable to determine if it should do the remediation at all. If `mount_has_to_exist == True AND fstab entry does not exist`, it will terminate the remediation, since not having a fstab entry is a passable state when mount_has_to_exist == True.

Switching to the proposed name makes the OVAL more clear and I think it also improves the remediation (though the description would have to be rewritten):
- not do anything if `fstab_entry_is_optional == True AND fstab entry does not exist`
- add a new entry if `fstab_entry_is_optional == False AND fstab entry does not exist`
- modify existing entry if `fstab_entry_is_optional == True/False AND fstab entry exists`

That said, I'm not super convinced about this change after thinking about it, so I'll close the PR until something more elegant comes along.